test/main.test.js

```diff
@@ -129,4 +129,40 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('handles concurrent then() calls with when v3.7.7 promises & bind option',
+  function() {
+    var timeout = 25;
+    function runWithPushedValue(pushedValue, bind, expectToFail) {
+      return new Promise(function concurrentExecution(outerResolve, reject) {
+        LoopBackContext.runInContext(function pushToContext() {
+          var ctx = LoopBackContext.getCurrentContext({bind: bind});
+          expect(ctx).is.an('object');
+          ctx.set('test-key', pushedValue);
+          var whenPromise = whenV377.promise(function(resolve) {
+            setTimeout(resolve, timeout);
```

Comment:
Can we simplify this code by using `promise.delay`? See https://github.com/cujojs/when/blob/master/docs/api.md#promisedelay